src/battle_script_commands.c

```diff
@@ -16283,3 +16258,41 @@ void BS_TrySetOctolock(void)
         gBattlescriptCurrInstr = cmd->nextInstr;
     }
 }
+
+void BS_SetGlaiveRush(void)
+{
+    NATIVE_ARGS();
+
+    gStatuses4[gBattlerAttacker] |= STATUS4_GLAIVE_RUSH;
+
+    gBattlescriptCurrInstr = cmd->nextInstr;
+}
+
+void BS_TryRelicSong(void)
+{
+    NATIVE_ARGS();
+
+    if (GetBattlerAbility(gBattlerAttacker) != ABILITY_SHEER_FORCE && !(gBattleMons[gBattlerAttacker].status2 & STATUS2_TRANSFORMED))
+    {
+        if (gBattleMons[gBattlerAttacker].species == SPECIES_MELOETTA_ARIA)
+            gBattleMons[gBattlerAttacker].species = SPECIES_MELOETTA_PIROUETTE;
+        else if (gBattleMons[gBattlerAttacker].species == SPECIES_MELOETTA_PIROUETTE)
+            gBattleMons[gBattlerAttacker].species = SPECIES_MELOETTA_ARIA;
+
+        BattleScriptPush(cmd->nextInstr);
+        gBattlescriptCurrInstr = BattleScript_AttackerFormChangeMoveEffect;
+    }
+    else
+        gBattlescriptCurrInstr = cmd->nextInstr;
+}
+
+void BS_ApplyPlasmaFists(void)
+{
+    NATIVE_ARGS();
+
+    u8 i;
+    for (i = 0; i < gBattlersCount; i++)
+        gStatuses4[i] |= STATUS4_PLASMA_FISTS;
```

Comment:
I can see you just moved the existing code, but whoever did it originally handled it in a weird way. Plasma Fists really should set an Ion Deluge (a field effect like Gravity and Fairy Lock) with a timer of 1 that's decremented at the end of the turn. It shouldn't be attached to the battlers on the field as a status.

Reply:
Makes sense. I'll change it to a field effect.

Reply:
`STATUS_FIELD_ION_DELUGE` was already present for the move so I used it for Plasma Fists as well.

Reply:
Looks better

Reply:
Great. Thx again!